Plugin/Holacash/Model/Payment.php

```diff
@@ -0,0 +1,201 @@
+        $this->checkoutCart = $checkoutCart;
+    }
+
+    /**
+     * @param  HolacashPayment $subject
+     * @param  $response
+     * @return null
+     * @throws \Magento\Framework\Exception\AlreadyExistsException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function beforeError(HolacashPayment $subject, $response)
+    {
+        $policyName = $this->purchaseHelper->getPolicyName(
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORES,
+            $this->storeManager->getStore()->getId()
+        );
+
+        $isPreAuth = $this->purchaseHelper->getIsPreAuth($policyName, 'holacash');
+
+        $quote = $this->checkoutCart->getQuote();
+
+        if ($isPreAuth === false || isset($quote) === false) {
+            return null;
+        }
+
+        $quoteId = $quote->getId();
+        /**
+         * @var $case \Signifyd\Connect\Model\Casedata
+        */
+        $case = $this->casedataFactory->create();
+        $this->casedataResourceModel->load($case, $quoteId, 'quote_id');
+
+        if ($case->isEmpty()) {
+            return null;
+        }
+
+        $message = $response["detail"]["message"] ?? 'Could not create charge';
+        $code = $response["detail"]["code"] ?? '';
+
+        switch ($code) {
+        case 'new_cvv_required':
+        case 'invalid_security_credentials':
+        case 'credit_card_validation_code_required':
+        case 'credit_card_validation_code_invalid':
+            $signifydReason = 'INVALID_CVC';
+            break;
+        case 'blocked_by_hola_cash_fraud_detection':
+        case 'invalid_anti_fraud_header_multiple':
+        case 'invalid_anti_fraud_header_format':
+        case 'invalid_anti_fraud_header_missing_ip_address':
+        case 'invalid_anti_fraud_header_missing_device_id':
+        case 'invalid_anti_fraud_header_missing_user_timezone':
+        case 'invalid_anti_fraud_header_invalid_user_timezone':
+        case 'invalid_anti_fraud_header_invalid_ip_address':
+        case 'three_ds_authentication_failed':
+        case 'credit_card_fraud_detected':
+        case '3D_authentication_failed':
+            $signifydReason = 'FRAUD_DECLINE';
+            break;
+        case 'bank_account_clabe_unrecognizable':
+        case 'bank_account_clabe_already_exists':
+        case 'bank_account_clabe_associated_with_hola_cash':
+        case 'bank_account_clabe_not_supported_spei_with_hola_cash':
+            $signifydReason = 'INCORRECT_NUMBER';
+            break;
+        case 'transaction_pin_required':
+        case 'credit_card_purchase_not_supported':
+        case 'credit_card_not_supported_on_line':
+        case 'credit_card_call_bank_to_authorize':
+        case 'card_declined':
+            $signifydReason = 'CARD_DECLINED';
+            break;
+        case 'credit_card_expired_year':
+        case 'credit_card_expired_month':
+        case 'credit_card_expired':
+        case 'invalid_card_expiry':
```

Comment:
@jpaliwal please indent the `case` so ti shows one idendentation level more than the `switch`